python/ray/client_builder.py

```diff
@@ -0,0 +1,97 @@
+import os
+import importlib
+import logging
+from dataclasses import dataclass
+from urllib.parse import urlparse
+from typing import Any, Dict, Optional, Tuple
+
+from ray.ray_constants import RAY_ADDRESS_ENVIRONMENT_VARIABLE
+from ray.job_config import JobConfig
+import ray.util.client_connect
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ClientInfo:
+    """
+    Basic information of the remote server for a given Ray Client connection.
+    """
+    dashboard_url: Optional[str]
+    python_version: str
+    ray_version: str
+    ray_commit: str
+    protocol_version: str
+
+
+class ClientBuilder:
+    """
+    Builder for a Ray Client connection.
+    """
+
+    def __init__(self, address: Optional[str]) -> None:
+        self.address = address
+        self.job_config = JobConfig()
+
+    def env(self, env: Dict[str, Any]) -> "ClientBuilder":
+        """
+        Set an environment for the session.
+        """
+        self.job_config = JobConfig(runtime_env=env)
+        return self
+
+    def connect(self) -> ClientInfo:
+        """
+        Begin a connection to the address passed in via ray.client(...).
+        """
+        client_info_tuple = ray.util.client_connect.connect(
```

Comment:
Looks like the tuple is actually a dict :P

Reply:
renaming to `client_info_dict`